packages/core/pluggableElementTypes/models/BaseViewModel.ts

```diff
@@ -14,6 +15,18 @@ const BaseViewModel = types
     menuItems(): MenuItem[] {
       return []
     },
+    currentLocation(): string | undefined {
```

Comment:
I think maybe this should be a getter.

```suggestion
    get currentLocation(): string | undefined {
```

Reply:
👍